advocacy_docs/edb-postgres-ai/migration-etl/data-migration-service/getting_started/prepare_schema.mdx

```diff
@@ -0,0 +1,68 @@
+---
+title: "Preparing and importing the schema"
+deepToC: true
+---
+
+Before you use EDB Data Migration Service (EDB DMS) to configure a data migration, you must prepare and import your schema to the target database. 
+
+Some of your schema's constraints must be included before the data migration takes place, whereas others must be applied [after the data migration is completed](apply_constraints). This ensures you can migrate without performance degradation. 
+
+## Schema integrity and performance considerations
+
+The presence of target database constraints, triggers, and WAL logging can impact the data migration performance. When possible, EDB recommends a two-step import of schema constraints.
+
+### `PRIMARY KEY` and `UNIQUE` constraints
+
+`PRIMARY KEY` and `UNIQUE` constraints are leveraged by EDB DMS to provide an exactly-once delivery when migrating data to the target database. Therefore, `PRIMARY KEY` and `UNIQUE` constraints should be included in the schema import that you perform before the data migration begins. Other types of constraints should be excluded from the schema import. 
+
+For rows in tables that do not have `PRIMARY KEY` or `UNIQUE` constraints it is only possible to achieve at-least-once delivery. Deduplication can be performed during the [data migration verification](verify_migration).
+
+!!!note 
+   `NOT NULL` constraints don't represent a significant performance impact for destination servers and can also be included in the schema import.
+!!!
+
+### `FOREIGN KEY`, `REFERENCES`, `CHECK`, `CASCADE` and `EXCLUDE` constraints
```

Comment:
@mlewandowski-edb - just to be clear on what you're suggesting...

1. Can we just drop `REFERENCES` and `CASCADE` from being mentioned then as they are configuration keywords for `FOREIGN KEY` constraints?

2. Are you just suggesting that I replace references to `EXCLUDE` with `EXCLUSION`?

Reply:
Where "`EXCLUDE` constraint" is used, yes I am suggesting that those be changed to "`EXCLUSION` constraint"

Reply:
Also, regarding `REFERENCES` and `CASCADE`, those terms don't need to be mentioned just because they are keywords used to configure a `FOREIGN KEY` constraint.